### DISCUSSION TITLE
memory-deny-write-execute trouble with late started threads

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#5118] [Q&A] [answered]
memory-deny-write-execute works well with lots of apps thumbs up.

But,  on some apps where threads starting not immediately after first daemon like keydb key values stores, because threads starting late after 7 seconds, the Keydb main process cant create threads with  memory-deny-write-execute enabled. 

Do you think its possible to do sleep timer for starting memory-deny-write-execute, when process starting successfully with all threads? 

Like

memory-deny-write-execute 7s/7000ms

Thanks and

best regards

**Comment by rusty-snake** [comment, marked as answer]
No. Not in a secure way.

**Reply by osevan** [reply]
Ok, are you sure?

**Reply by rusty-snake** [reply]
To load a seccomp filter after some time, the program it self must load it (via LD_PRELOAD library) which is of course not secure or to somehow load it into an other program (`ptrace`) which is difficult because of yama, ... .

**Reply by osevan** [reply]
OK thank you

**Reply by rusty-snake** [reply]
Update on this, technically it would be possible with seccomp user-space notifications.
